src/Workspaces/Remote/ServiceHub/Services/BrokeredServiceBase.cs

```diff
@@ -106,5 +110,39 @@ protected async ValueTask RunServiceAsync(Func<CancellationToken, ValueTask> imp
             cancellationToken = source.Token;
             return source;
         }
+
+        private static bool ReportUnexpectedException(Exception exception, CancellationToken cancellationToken)
+        {
+            if (exception is OperationCanceledException)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    return false;
+                }
+
+#if DEBUG
+                // Catch and process an unexpected OCE in the exception handler.
+                // We propagate the OCE in release builds to avoid unnecessary service failures
+                // since most of our code base handles OCE regardless of the token.
+                return FatalError.ReportWithoutCrash(exception);
+#endif
+            }
+
+            return FatalError.ReportWithoutCrashAndPropagate(exception);
+        }
+
+        [DoesNotReturn]
+        private static void OnUnexpectedException(Exception exception, CancellationToken cancellationToken)
```

Comment:
❔ What's the harm in leaving this out. I'm having a really hard time understanding what it's adding.

Reply:
Removed.